spec/services/hearing_schedule/generate_hearing_days_schedule_spec.rb

```diff
@@ -337,5 +288,220 @@
         expect(subject.count).to eq(22)
       end
     end
+
+    # Travel board days don't have hearing days they are blackout dates
+    context "combined allocation (video, virtual, central)" do
+      # Take a list of hearing_days and condense them per ro, date, and type,
+      # results look like this.
+      # {"RO39" => {
+      #   "day_counts" => {
+      #     "08-18-2021" => {
+      #       "video" => 12,
+      #       "virtual" => 41,
+      #       "central" => 1}
+      #     }
+      #   }
+      # }
+      def condense_hearing_days(hearing_days)
+        # Generate day_counts for each date/type from the hearing_days
+        day_counts = {}
+        hearing_days.each do |hearing_day|
+          # Extract the keys we will use to create nested hashes
+          scheduled_for = hearing_day[:scheduled_for]
+          type = hearing_day[:request_type]
+          # Deal with central hearings by creating "C" as an RO (which it's not really)
+          ro = (hearing_day[:request_type] == "C") ? "C" : hearing_day[:regional_office]
+
+          # Create the hashes/keys if they don't exist
+          day_counts[ro] = {} unless day_counts.key?(ro)
+          day_counts[ro][scheduled_for] = {} unless day_counts[ro].key?(scheduled_for)
+          day_counts[ro][scheduled_for][type] = 0 unless day_counts[ro][scheduled_for].key?(type)
+
+          # Update the count for this ro, date, type combo
+          day_counts[ro][scheduled_for][type] += 1
+        end
+        day_counts
+      end
+
+      # If count < min then update min, if count > max update max
+      # If max is nil, set it to count
+      # if min is nil, set it to count
+      def update_min_max(count, current_min, current_max)
+        # If count is nil, update nothing.
```

Comment:
Since we're just returning, i'd prefer that language.
```suggestion
        # If count is nil, return nothing.
```